### ISSUE TITLE
"parse_build_args.py" file parser arguments are not accessible anymore in "os.environ".

### ISSUE THREAD

**Issue by KelSolaar** [nerdvegas/rez#590]
Hi,

I'm trying to update Rez from 2.16.0 to 2.28.0 and one of the first thing I noticed is my builds failing :)

We create `parse_build_args.py` files to support some custom build arguments with content for example as follows:

```python
# -*- coding: utf-8 -*-
parser.add_argument(
    '--skip-code-formatting',
    action='store_true',
    help='Skips code formatting.')
```

and then in the build file (we use Fabric, thus `fabfile.py`), we do things like that:

```python
    if not int(os.environ['__PARSE_ARG_SKIP_CODE_FORMATTING']):
        code_formatting()
```

Unfortunately the `__PARSE_ARG_SKIP_CODE_FORMATTING` is nowhere to be seen now.

Cheers,

Thomas

**Comment by instinct-vfx** [comment]
Hey Thomas,

We ran into this too. Here is my report: https://github.com/nerdvegas/rez/issues/587

An here is a commit fixing this. PR is coming soon (Thanks Iraklis!) https://github.com/lambdaclan/rez/commit/af81b2492d297ad3350d1988bca47de0b4dcae7c

**Comment by KelSolaar** [comment]
@instinct-vfx : Ah good to know, will follow closely the future PR!

**Comment by KelSolaar** [comment]
I made a custom build with two cherry-picked commits from @lambdaclan and it fixed the issue for me.

Would be fantastic to have a PR for that so that we don't have to deploy a custom version, nothing really bad about it but it is extra maintenance steps.

Cheers,

Thomas

**Comment by nerdvegas** [comment]
I'm keeping an eye out for this PR also. There will be another round of
merges coming soon and this one's on the list.
A

**Comment by lambdaclan** [comment]
@KelSolaar  @nerdvegas I have just added a PR for the issue +1 more incoming!

**Comment by nerdvegas** [comment]
I believe this is now fixed in 2.30.0, pls reopen if incorrect
